From a1df649db70665169dd1a83a4b8f7f881ac704e3 Mon Sep 17 00:00:00 2001
From: Hugues Fruchet <hugues.fruchet@foss.st.com>
Date: Thu, 11 May 2023 18:13:08 +0200
Subject: [PATCH 0326/1141] media: hantro: add NV12 encode support for
 STM32MP25 VENC

Add support of encoding from NV12 single-plane frame.

Signed-off-by: Hugues Fruchet <hugues.fruchet@foss.st.com>
Change-Id: I036f7ec25466529d447503b11bf38db555dd1912
---
 .../media/platform/verisilicon/stm32mp25_venc_hw.c  | 13 +++++++++++++
 1 file changed, 13 insertions(+)

--- a/drivers/media/platform/verisilicon/stm32mp25_venc_hw.c
+++ b/drivers/media/platform/verisilicon/stm32mp25_venc_hw.c
@@ -22,6 +22,19 @@
 
 static const struct hantro_fmt stm32mp25_venc_fmts[] = {
 	{
+		.fourcc = V4L2_PIX_FMT_NV12,
+		.codec_mode = HANTRO_MODE_NONE,
+		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_YUV420SP,
+		.frmsize = {
+			.min_width = 96,
+			.max_width = 8176,
+			.step_width = MB_DIM,
+			.min_height = 32,
+			.max_height = 8176,
+			.step_height = MB_DIM,
+		},
+	},
+	{
 		.fourcc = V4L2_PIX_FMT_YUV420M,
 		.codec_mode = HANTRO_MODE_NONE,
 		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_YUV420P,
